### PR TITLE
Added new module ForecastData

### PR DESCRIPTION
**Pull request by Vagik**

### CODE REVIEW

File: ForecastData/src/main/java/com/ait/forecastdata/IOpenWeather.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.ait.forecastdata
+
+import com.ait.forecastdata.models.weather.openWeather.OWCommonWeatherInfo
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface IOpenWeather {
+    @GET("data/2.5/onecall?units=metric&exclude=minutely,hourly,alerts&appid=fb5b8d7fc64091a58eb0247d3913b6ae")
```
Comment (Vagik) [inline ForecastData/src/main/java/com/ait/forecastdata/IOpenWeather.kt]: Need to discuss, how to hide the appid value

Comment (Vagik) [reply ForecastData/src/main/java/com/ait/forecastdata/IOpenWeather.kt]: Will be moved to assets

File: app/src/main/java/com/ait/alphaforecast/OWAppIdProvider.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.ait.alphaforecast
+
+import android.app.Application
+import com.ait.dataapi.AppIdProvider
+
+class OWAppIdProvider(private val application: Application): AppIdProvider {
+    override fun getAppId(): String {
+        val bufferReader = application.assets
```
Comment (ConstAlpha) [inline app/src/main/java/com/ait/alphaforecast/OWAppIdProvider.kt]: Mb better to use ```AssetManager```?

File: app/src/main/java/com/ait/alphaforecast/MainActivity.kt [hunk 3]
```diff
@@ -2,12 +2,23 @@ package com.ait.alphaforecast
...
+import kotlinx.coroutines.launch
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_main)
+
+        lifecycleScope.launch(Dispatchers.Main) {
```
Comment (ConstAlpha) [inline app/src/main/java/com/ait/alphaforecast/MainActivity.kt]: Please remove this. We don't need any code in our Activity (except of ui setting up)

File: data-api/src/main/java/com/ait/dataapi/AppIdProvider.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.ait.dataapi
```
Comment (ConstAlpha) [inline data-api/src/main/java/com/ait/dataapi/AppIdProvider.kt]: Use ```com.ait.data.api``` package name here and in other places

Comment (ConstAlpha) [reply data-api/src/main/java/com/ait/dataapi/AppIdProvider.kt]: And ```com.ait.data.implementation``` in impl module

File: data-api/src/main/java/com/ait/dataapi/model/ForecastSource.kt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.ait.dataapi.model
+
+enum class ForecastSource {
+    OPEN_WEATHER,
+    CHTO_TO_ESHYO
```
Comment (ConstAlpha) [inline data-api/src/main/java/com/ait/dataapi/model/ForecastSource.kt]: =)

File: data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import com.ait.dataimplementation.service.template.TemplateDataService
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+
+class WeatherForecastManager(private val appIdProvider: AppIdProvider, private val source: ForecastSource) {
```
Comment (ConstAlpha) [inline data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt]: Seems this class should  be renamed to ```WeatherForecastRepository```

File: data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    suspend fun getWeatherInfo(latitude: Double, longitude: Double): ForecastedWeatherInfo? {
+
+        // TODO: Will be reworked later with Hilt
+        val forecastService: ForecastService = when (source) {
+            ForecastSource.OPEN_WEATHER -> {
+                val client = OkHttpClient().newBuilder()
```
Comment (ConstAlpha) [inline data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt]: There must be another class for api building. Just pass ForecastService interface in constructor

Comment (ConstAlpha) [reply data-implementation/src/main/java/com/ait/dataimplementation/WeatherForecastManager.kt]: We must follow single responsibility rule

File: app/src/main/java/com/ait/alphaforecast/OWAppIdProvider.kt [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.ait.alphaforecast
+
+import android.app.Application
+import com.ait.dataapi.AppIdProvider
+
+class OWAppIdProvider(private val application: Application): AppIdProvider {
```
Comment (ConstAlpha) [inline app/src/main/java/com/ait/alphaforecast/OWAppIdProvider.kt]: Please use PascalCase notation here and in other places (rename to ```OwAppIdProvider```)

**Review by ConstAlpha — CHANGES_REQUESTED** [review]

**Pull request merged**